Puma::ControlCLI - allow refork command to be sent as a request

MSP-Greg opened the pull request:
### Description

Currently, the refork command will only be sent as a signal by `Puma::ControlCLI`.  Change to allow it be sent as a request if the control server is setup with a uri.

Closes #2866

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [ ] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

nateberkopec commented on lib/puma/control_cli.rb:
```diff
@@ -33,10 +33,10 @@ class ControlCLI
 
     # commands that cannot be used in a request
-    NO_REQ_COMMANDS = %w{refork}.freeze
+    NO_REQ_COMMANDS = %w[].freeze
```
Probably should just remove this constant then, yes?

nateberkopec replied:
I feel like I'm missing something.

MSP-Greg replied:
I can, but I thought I'd leave the code in case a 'signal only' command was added.  Or, previous to this PR, `refork` was a signal only command, and I never tested it using a control server that had a control uri.

Leaving as is means that CI will check that if one is added.  Ok, I know it's a little odd...

MSP-Greg replied:
I commented out the constant and the clause that used it.  Same for the test (skipped if the constant is undefined).

I can remove if you want...  I just don't want myself or someone else making the same mistake again, so the commented code leave a **big** hint...

johnnyshields commented on lib/puma/control_cli.rb:
```diff
@@ -32,11 +32,13 @@ class ControlCLI
+    #
     # commands that cannot be used in a request
-    NO_REQ_COMMANDS = %w{refork}.freeze
+    # NO_REQ_COMMANDS = %w[].freeze
```
Should just remove this altogether?

MSP-Greg commented:
I rebased and added 'signal only' commands that were previously unavailable using `ControlCLI`. See https://github.com/puma/puma/pull/2868/commits/eb87fe59dc4098c1add5f2f9b19027fe1ae8b91b

I think I found all of them.  Hence, `ControlCLI` can be used to send all signal commands that work with the main Puma process.
